var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -30,14 +30,15 @@ class PySetuptools(PythonPackage):
        upgrading, installing, and uninstalling Python packages."""
 
     homepage = "https://pypi.python.org/pypi/setuptools"
-    url      = "https://pypi.python.org/packages/source/s/setuptools/setuptools-11.3.tar.gz"
+    url      = "https://pypi.io/packages/source/s/setuptools/setuptools-25.2.0.tar.gz"
 
-    version('25.2.0', 'a0dbb65889c46214c691f6c516cf959c',
-            url="https://pypi.python.org/packages/9f/32/81c324675725d78e7f6da777483a3453611a427db0145dfb878940469692/setuptools-25.2.0.tar.gz")
+    version('34.2.0', '41b630da4ea6cfa5894d9eb3142922be',
+            url="https://pypi.io/packages/source/s/setuptools/setuptools-34.2.0.zip")
```

Comment:
Started digging through the `CHANGES.rst` for `setuptools`:
```
v34.0.0
-------

* #581: Instead of vendoring the growing list of
  dependencies that Setuptools requires to function,
  Setuptools now requires these dependencies just like
  any other project. Unlike other projects, however,
  Setuptools cannot rely on ``setup_requires`` to
  demand the dependencies it needs to install because
  its own machinery would be necessary to pull those
  dependencies if not present (a bootstrapping problem).
  As a result, Setuptools no longer supports self upgrade or
  installation in the general case. Instead, users are
  directed to use pip to install and upgrade using the
  ``wheel`` distributions of setuptools.

  Users are welcome to contrive other means to install
  or upgrade Setuptools using other means, such as
  pre-installing the Setuptools dependencies with pip
  or a bespoke bootstrap tool, but such usage is not
  recommended and is not supported.

  As discovered in #940, not all versions of pip will
  successfully install Setuptools from its pre-built
  wheel. If you encounter issues with "No module named
  six" or "No module named packaging", especially
  following a line "Running setup.py egg_info for package
  setuptools", then your pip is not new enough.

  There's an additional issue in pip where setuptools
  is upgraded concurrently with other source packages,
  described in pip #4253. The proposed workaround is to
  always upgrade Setuptools first prior to upgrading
  other packages that would upgrade Setuptools.
```
From what I understand, this means that `setuptools` doesn't care whether or not we can bootstrap it. I think I can hack `appdirs` to make the installation work, but let's hope this behavior doesn't continue.